feat: LEAP-2036: Summary page for View All

hlomzik opened the pull request:
Create a special Summary page inside View All view to display all labelings across all annotations in current task.
Here are the `LabelingSummary` and `DataSummary` components to display info about control tags and object tags respectively.

hlomzik committed:
feat: LEAP-2036: Summary page for View All

Create a special Summary page inside View All view to display all labelings
across all annotations in current task.
Here are the LabelingSummary and DataSummary components to display info
about control tags and object tags respectively.

hlomzik committed:
Fix types and user info displaying

Proper user info for Userpic and `user` in store.
Proper types for them.

hlomzik committed:
Display Summary instead of Grid for now

hlomzik committed:
Add Number and Choices to supported labeling types

hlomzik committed:
Improve styles of user/id part

hlomzik committed:
Don't display not submitted drafts in Summary

nass600 commented on web/libs/editor/src/components/App/App.jsx:
```diff
@@ -47,7 +47,8 @@ import { ToastProvider, ToastViewport } from "@humansignal/ui/lib/toast/toast";
 import Debug from "../Debug";
-import Grid from "./Grid";
+// import Grid from "./Grid";
+import Summary from "../Summary/Summary";
```
I would use a more precise name for this component given it is pretty open, maybe `AnnotationSummary` or `ReviewSummary` or anything else that gives a bit more information

hlomzik replied:
thank you, good point! renamed to `TaskSummary`

nass600 commented on web/libs/editor/src/components/Summary/DataSummary.tsx:
```diff
@@ -0,0 +1,32 @@
+    <table className="table-auto w-full">
+      <thead>
+        <tr>
+          {Object.entries(data_types).map(([field, type]) => (
```
I would add a check here so we can safely assume we will have `data_types`

nass600 commented on web/libs/editor/src/components/Summary/LabelingSummary.tsx:
```diff
@@ -0,0 +1,114 @@
+
+export const LabelingSummary = ({ annotations, controls }: Props) => {
+  return (
+    <table className="w-full">
```
Unless specifically needed I would use Grid over tables so we can easily ensure a responsive design

hlomzik replied:
this will be fixed in follow-up PR, I'll rewrite this table to react-table

hlomzik committed:
Add new FF for the Summary

hlomzik committed:
Add umbrella component ViewAll

It chooses between tabbed view with FF ON and usual Grid with FF OFF.
Tabbed view has new Summary tab and old Grid tab.
Selected tab is preserved between sessions.

hlomzik committed:
Small fixes

- no tooltip on avatar, we already show names
- annotation tabs are always displayed with new Summary

hlomzik committed:
Improve labeling types

- Add TextArea tag
- Add support for per-regions
  - for now per-region flag is calculated by simple rule: >1 results -> per-region
- Move renderers to a separate file
- Allow to return null to fallback to default text

hlomzik committed:
Improve styles and show Number cards

More semantic styles.
Local Card component.
Numbers on the top.

hlomzik committed:
Move Labels renderer to a function and reuse it

All labels will be displayed the same, so just reuse the same renderer.

hlomzik committed:
Add missing type

hlomzik committed:
Small fixes + respect new "annotations:summary" interface

Gondragos commented on web/libs/editor/src/components/App/ViewAll.tsx:
```diff
@@ -0,0 +1,49 @@
+export const ViewAll = ({ store, annotations, root }: Props) => {
+  const [tab, setTab] = usePersistentState<"summary" | "compare">("view-all-tab", "summary");
+
+  if (ff.isActive(ff.FF_SUMMARY)) {
```
Should there also be a check that it is not standalone mode? I mean at least `window.DM` check, but maybe there is a better way.

hlomzik committed:
Use submitted results not the current draft

Agreement is calculated on backend over submitted annotations.
So displayed results should also be only submitted once.
We have `versions` with `draft` and `result`, so we should use the latter.

hlomzik committed:
Merge branch 'develop' into fb-leap-2036/summary

hlomzik committed:
Use LSF tags info; small improvements; linting

Get (almost) all info about tags from LSF; don't use DM.
The only lefover is agreement, should be fixed soon.
Fix all the related code to work with real tags.
Add new types to describe store and tags.
Fix linting here and there.

hlomzik committed:
Add `agreement` as a field to task model

This automatically accepts agreement passed via `task` from outer sources.

hlomzik commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14956470496)
> Successfully merged:  rename web/libs/datamanager/src/components/App/{App.jsx => App.tsx} (73%)

hlomzik committed:
Merge branch 'develop' into 'fb-leap-2036/summary'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14956470496

hlomzik committed:
Display predictions properly

hlomzik committed:
Merge branch 'fb-leap-2036/summary' of github.com:HumanSignal/label-studio into fb-leap-2036/summary

hlomzik committed:
Merge branch 'develop' into fb-leap-2036/summary

hlomzik committed:
Display actual data for every object tag

TimeSeries still show `$ts` instead of real value.
But real value will be a huge csv, so will fix this later.

hlomzik committed:
Rename Summary to TaskSummary

Gondragos commented on web/libs/editor/src/components/App/ViewAll.tsx:
```diff
@@ -0,0 +1,49 @@
+};
+
+export const ViewAll = ({ store: annotationStore, annotations, root }: Props) => {
+  const [tab, setTab] = usePersistentState<"summary" | "compare">("view-all-tab", "summary");
```
`"summary"` and `"compare"` could be defined as constants or something like that.

Gondragos commented on web/libs/editor/src/stores/types.d.ts:
```diff
@@ -10,24 +11,42 @@ type MSTResult = {
+type MSTObjectTag = {
   isReady?: boolean;
+  name: string;
+  isControlTag: false;
+  isObjectTag: true;
```
Most likely, both fields are not present in the tag at the same time.

hlomzik commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/15355888113)
> Successfully merged:  create mode 100644 label_studio/tasks/tests/test_api.py

hlomzik committed:
Merge branch 'develop' into 'fb-leap-2036/summary'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15355888113
